flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -313,11 +296,15 @@ private <T> Optional<T> getLatestPriorProperty(Function<ArchivedExecution, T> ex
      * @return The latest prior execution location, or null, if there is none, yet.
      */
     public Optional<TaskManagerLocation> findLatestPriorLocation() {
-        return getLatestPriorProperty(ArchivedExecution::getAssignedResourceLocation);
+        return Optional.ofNullable(lastAssignedLocation);
+    }
+
+    void setLatestPriorAllocation(AllocationID lastAssignedAllocationID) {
```

Comment:
I thought about it once more: I think it makes sense to merge `setLatestPriorAllocation` and `setLatestPrioirLocation` into something like `setLatestPriorSlotAssignment(TaskManagerLocation, AllocationID)` because both are closely related...